docs/api/error-handling.md

```diff
@@ -0,0 +1,164 @@
+---
+id: error-handling
+title: Error handling
+sidebar_label: Error handling
+---
+
+Tartiflette Engine provides an unique `TartifletteError` class, which is designed to format errors accordingly to the GraphQL specification.
+
+When an error occurs from both tartiflette engine and your resolvers, it is formated to respect the regular GraphQL Error shape.
+
+```json
+{
+    "data": {
+        "myField": {}
+    },
+    "errors": [
+        {
+            "message": "Something wrong happened.",
+            "path": ["myField"],
+            "locations": [
+                {
+                    "line": 1,
+                    "column": 2
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Default use-case
+
+Commonly, each exception is wrapped into a `TartifletteError` exception, the message is copied, and the original exception is stored into theinstance is put into the `original_error` property.
+
+In that use-case, there is no way to specify custom properties.
+
+e.g
+
+```python
+from tartiflette import Resolver
+
+@Resolver("Query.myField")
+async def resolve_my_field(parent, args, ctx, info):
+    # bla bla bla
+    raise Exception("There is an error with your storage")
+
+```
+
+Will return this response payload.
+
+
+```json
+{
+    "data": {
+        "myField": {}
+    },
+    "errors": [
+        {
+            "message": "Something wrong happened.",
+            "path": ["myField"],
+            "locations": [
+                {
+                    "line": 1,
+                    "column": 2
+                }
+            ]
+        }
+    ]
+}
+```
+
+## Add custom properties to the response payload
+
+In some use-cases, like form's validation, functional error, you want to expose more fine-grained details to your API's clients. These details allow them to adapt their UI on your custom attributes. 
+
+[As explained in the specification](https://github.com/graphql/graphql-spec/blob/master/spec/Section%207%20--%20Response.md#errors), this is possible to add extra properties to the response error payload. They have to be included into the `extensions` property of your error.
```

Comment:
Can you link the https://graphql.github.io/graphql-spec/June2018/#sec-Errors site instead of the master code (that will move, we are aiming at the 06/2018 spec)